### PR TITLE
OSVR-Unity RenderManager

### PR DESCRIPTION
**Pull request by DuFF14**
Adds RenderManager functionality to OSVR-Unity (nVidia Direct mode, timewarp, distortion).
Will require Unity 5.2+ to use RenderManager features.
Also requires osvrUnityRenderingPlugin.dll, osvrRenderManager.dll, SDL2.dll, glew32.dll to be added to Plugins folders.

### CODE REVIEW

File: OSVR-Unity/Assets/OSVRUnity/src/OsvrRenderManager.cs [hunk 1]
```diff
@@ -0,0 +1,254 @@
...
+            private static extern Byte CreateRenderManagerFromUnity(OSVR.ClientKit.SafeClientContextHandle /*OSVR_ClientContext*/ ctx);
+
+            [StructLayout(LayoutKind.Sequential)]
+            public struct OSVR_ViewportDescription
+            {
+                public double left;    //< Left side of the viewport in pixels
+                public double lower;   //< First pixel in the viewport at the bottom.
+                public double width;   //< Last pixel in the viewport at the top
+                public double height;   //< Last pixel on the right of the viewport in pixels
+            }
```
Comment (JeroMiya) [inline OSVR-Unity/Assets/OSVRUnity/src/OsvrRenderManager.cs]: Are these supposed to be doubles? We have a ```Viewport``` struct in Managed-OSVR that uses ```int32``` fields.

File: OSVR-Unity/Assets/OSVRUnity/src/Math.cs [hunk 2]
```diff
@@ -45,6 +45,11 @@ public static Quaternion ConvertOrientation(OSVR.ClientKit.Quaternion quat)
                 /// Wikipedia may say quaternions are not handed, but these needed modification.
                 return new Quaternion(-(float)quat.x, -(float)quat.y, (float)quat.z, (float)quat.w);
             }
+            public static Quaternion ConvertOrientationFromRenderManager(OSVR.ClientKit.Quaternion quat)
```
Comment (JeroMiya) [inline OSVR-Unity/Assets/OSVRUnity/src/Math.cs]: Just curious - why are orientations from render manager different than those from OSVR?

**Comment by JeroMiya** [comment]
This looks good. I had a couple of questions, but this should be OK to merge.

File: OSVR-Unity/Assets/OSVRUnity/src/OsvrRenderManager.cs (hunk 1 above)
Comment (DuFF14) [inline OSVR-Unity/Assets/OSVRUnity/src/OsvrRenderManager.cs]: Didn't find any way around this. Had to create this struct to get the viewport as it comes from RenderManager. I do convert that to an OSVR.ClientKit.Viewport here: https://github.com/OSVR/OSVR-Unity/blob/OsvrUnity_RenderManager/OSVR-Unity/Assets/OSVRUnity/src/OsvrRenderManager.cs#L128

Open to suggestions if there's a better way to handle this.

File: OSVR-Unity/Assets/OSVRUnity/src/Math.cs (hunk 2 above)
Comment (DuFF14) [inline OSVR-Unity/Assets/OSVRUnity/src/Math.cs]: I'm not sure. Perhaps @russell-taylor has a good answer.

Comment (russell-taylor) [inline OSVR-Unity/Assets/OSVRUnity/src/Math.cs]: I didn't know that they were.  With RenderManager, you can add a room-to-world transform and make them whatever you want, but that has now also been added to Core with the reset-yaw code.

The Core transforms and RenderManager transforms were developed independently, so may have different coordinate systems.  But OSVR defines a standard, so if they are different then one of them is wrong.

File: OSVR-Unity/Assets/OSVRUnity/src/OsvrRenderManager.cs (hunk 1 above)
Comment (russell-taylor) [inline OSVR-Unity/Assets/OSVRUnity/src/OsvrRenderManager.cs]: I think the fix is to get RenderManager and Core to use the same structure, so will file a separate issue.  Ryan and I can fight about which way the structure is.

**Comment by DuFF14** [comment]
Thanks for the review. Merging.

**Pull request merged**